src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs

```diff
@@ -42,44 +59,64 @@ public static IEndpointConventionBuilder MapIdentityApi<TUser>(this IEndpointRou
         {
             var userManager = sp.GetRequiredService<UserManager<TUser>>();
 
+            if (!userManager.SupportsUserEmail)
+            {
+                throw new NotSupportedException($"{nameof(MapIdentityApi)} requires a user store with email support.");
+            }
+
+            var userStore = sp.GetRequiredService<IUserStore<TUser>>();
+            var emailStore = (IUserEmailStore<TUser>)userStore;
+
             var user = new TUser();
-            await userManager.SetUserNameAsync(user, registration.Username);
+            await userStore.SetUserNameAsync(user, registration.Username, CancellationToken.None);
+            await emailStore.SetEmailAsync(user, registration.Email, CancellationToken.None);
             var result = await userManager.CreateAsync(user, registration.Password);
 
             if (result.Succeeded)
             {
+                await SendConfirmationEmailAsync(user, userManager, registration.Email);
                 return TypedResults.Ok();
             }
 
-            return TypedResults.ValidationProblem(result.Errors.ToDictionary(e => e.Code, e => new[] { e.Description }));
+            return CreateValidationProblem(result);
         });
 
-        routeGroup.MapPost("/login", async Task<Results<UnauthorizedHttpResult, Ok<AccessTokenResponse>, SignInHttpResult>>
-            ([FromBody] LoginRequest login, [FromQuery] bool? cookieMode, [FromServices] IServiceProvider sp) =>
+        routeGroup.MapPost("/login", async Task<Results<Ok<AccessTokenResponse>, ProblemHttpResult, IResult>>
```

Comment:
Should we put `EmptyHtppResult` here instead of `IResult`? `IResult` basically means anything can be returned from this endpoint?

Reply:
It's not really empty. It's just not overwriting the AccessTokenResponse.